piet/src/text.rs

```diff
@@ -94,6 +135,26 @@ pub trait TextLayout {
     fn hit_test_text_position(&self, text_position: usize) -> Option<HitTestTextPosition>;
 }
 
+/// Metadata about each line in a text layout.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct LineMetric {
+    /// Index (in code units) of the start of the line, offset from the beginning of the text.
+    pub line_start_offset: usize,
+
+    /// Line length (in code units), but offset from the beginning of the text. So it's the length
+    /// of this line summed with the lengths of all the lines before it.
+    ///
+    /// Does not include trailing whitespace.
+    pub line_end_offset: usize,
+
+    /// Line width.
+    /// Does not include trailing whitespace.
+    pub cumulative_width: f64,
+
+    /// Length (in code units) of trailing whitespace.
```

Comment:
if this is code units it should be an integer type?

Reply:
I made some mixup here; I think trailing whitespace should be in code units (usize)